### PR TITLE
JSON db fix disable/remove 71636 columnsortorder

### PR DESCRIPTION
**Pull request by marcghaly**
Hi @safatshahin,
Here is the fix for the unremoved columns from db.

2 files were changed:

`admin/qbankplugins.php`
`lib/classes/plugininfo/qbank.php `

### CODE REVIEW

File: lib/classes/plugininfo/qbank.php [hunk 1]
```diff
@@ -130,4 +130,33 @@ public function load_settings(\part_of_admin_tree $adminroot, $parentnodename, $
...
+     *
+     * This is intended for disabling of plugin, some DB table purging, etc.
+     *
+     * NOTE: to be called from uninstall_plugin() only.
+     * @private
+     */
+    public function uninstall_cleanup() {
+        $plugintoremove = $this->displayname;
+        self::remove_unused_column_from_db($plugintoremove);
+    }
```
Comment (guillogo) [inline lib/classes/plugininfo/qbank.php:145]: Please call parent::pre_uninstall_cleanup() at the end `parent::uninstall_cleanup();`

File: lib/classes/plugininfo/qbank.php [hunk 2]
```diff
@@ -130,4 +130,33 @@ public function load_settings(\part_of_admin_tree $adminroot, $parentnodename, $
...
+
+    /**
+     * Removes any uninstalled or disabled plugin column in the config_plugins 'qbanksortorder' in core_question.
+     * @param string $plugintoremove Plugin displayed name to remove.
```
Comment (guillogo) [inline lib/classes/plugininfo/qbank.php:149]: Can you add a new line before @param thank you.

**Review by guillogo — CHANGES_REQUESTED** [review]
Hi @marcghaly just two small changes and is ready to go. Also I was thinking in how to handle when we install again a plugin but it will add it at the end so doesn't matter if we add it or not to the json. Cheers!

File: lib/classes/plugininfo/qbank.php (hunk 1 above)
Comment (marcghaly) [reply lib/classes/plugininfo/qbank.php:145]: Hi @guillogo looks like the method `pre_uninstall_cleanup()` does not exist in the parent class `lib/classes/plugininfo/base.php` should I call `uninstall_cleanup` instead ?

File: lib/classes/plugininfo/qbank.php (hunk 2 above)
Comment (marcghaly) [reply lib/classes/plugininfo/qbank.php:149]: added

**Pull request merged**